lib/statesman/adapters/active_record_queries.rb

```diff
@@ -39,7 +39,7 @@ def initialize(**args)
         end
 
         def included(base)
-          ensure_inheritance(base)
+          ensure_inheritance(base) unless base.subclasses.none?
```

Comment:
Generally think this reads better as an if as I don't have to double negate

```suggestion
          ensure_inheritance(base) if base.subclasses.any?
```

Also I think subclasses was only added in Ruby 3.1 so we may need to change this to support 3.x which is still currently supported

Reply:
> Generally think this reads better as an if as I don't have to double negate

Very good point 👍🏼 

> Also I think subclasses was only added in Ruby 3.1 

You're right: https://ruby-doc.org/core-3.1.0/Class.html#method-i-subclasses

`ActiveSupport` has a `DescendantsTracker` module which may be of help? It looks like it historically had a [`direct_descendants`](https://api.rubyonrails.org/v6.0.2.2/classes/ActiveSupport/DescendantsTracker.html#method-i-direct_descendants), but that method has since been removed from [`DescendantsTracker`](https://api.rubyonrails.org/classes/ActiveSupport/DescendantsTracker.html)

Reply:
Bit of a tricky one as Ruby 3.0 is EOL in ~3 months it does feel like something we could ignore relatively soon, can we just see if `base` responds to `subclasses` so it doesn't go 💥

Reply:
👍🏼  So are you thinking: 

```suggestion
          ensure_inheritance(base) if base.respond_to?(:subclasses) && base.subclasses.any?
```